### PR TITLE
Refactor initialization phase of transaction

### PR DESCRIPTION
**Pull request by dranikpg**
InitByArgs is 200+ LOC and I need to extend it to support different multi-transaction modes, so I decided to refactor it a little, else it'll become an unmanageable mess 

I've split it into multiple commits, each of them encompasses one or two changes

### CODE REVIEW

File: src/server/transaction.cc [hunk 1]
```diff
@@ -235,16 +139,127 @@ OpStatus Transaction::InitByArgs(DbIndex index, CmdArgList args) {
...
+    for (unsigned j = 0; j < reverse_index_.size(); ++j) {
+      reverse_index_[j] = j + key_index.start - 1;
+    }
+  }
+}
+
+/**
+ *
+ * There are 4 options that we consider here:
+ * a. T spans a single shard and its not multi.
```
Comment (romange) [inline src/server/transaction.cc:206]: use consistent notion for transaction: for example, `Tx` instead of T and Trans

**Review by romange — COMMENTED** [review]
My only ask here: please try to do as little functional changes as possible for the refactoring PR, 
and do all the functional changes in the subsequent PRs

File: src/server/transaction.cc (hunk 1 above)
Comment (dranikpg) [reply src/server/transaction.cc:206]: I didn't touch you comments at all yet

**Comment by dranikpg** [comment]
There are no functional changes in this PR

File: src/server/transaction.cc (hunk 1 above)
Comment (romange) [reply src/server/transaction.cc]: oh, I thought their are yours. in that case the comments are perfect 🙃

**Review by romange — APPROVED** [review]

**Pull request merged**